### PR TITLE
Add architecture graph for cri-containerd.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes-incubator/cri-containerd/issues/446.

The doc is extracted from http://blog.kubernetes.io/2017/11/containerd-container-runtime-options-kubernetes.html.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Let's use an example to demonstrate how cri-containerd works for the case when Kubelet creates a single-container pod:
+* Kubelet calls cri-containerd, via the CRI runtime service API, to create a pod;
+* cri-containerd uses containerd to create and start a special [pause container](https://www.ianlewis.org/en/almighty-pause-container) (the sandbox container) and put that container inside the pod’s cgroups and namespace (steps omitted for brevity);
+* cri-containerd configures the pod’s network namespace using CNI;
+* Kubelet subsequently calls cri-containerd, via the CRI image service API, to pull the application container image;
+* cri-containerd further uses containerd to pull the image if the image is not present on the node;
+* Kubelet then calls cri-containerd, via the CRI runtime service API, to create and start the application container inside the pod using the pulled container image;
+* cri-containerd finally calls containerd to create the application container, put it inside the pod’s cgroups and namespace, then to start the pod’s new application container.
+After these steps, a pod and its corresponding application container is created and running.
```
Comment (mikebrow) [inline docs/architecture.md]: looks good I'd add a link to containerd for more reading as well as a link to the kublet docs.

**Comment by yanxuean** [comment]
cool

File: docs/architecture.md (hunk 1 above)
Comment (Random-Liu) [reply docs/architecture.md]: Will do.

Comment (Random-Liu) [reply docs/architecture.md]: Done.

**Comment by Random-Liu** [comment]
Apply LGTM based on https://github.com/kubernetes-incubator/cri-containerd/pull/501#pullrequestreview-83604428

**Pull request merged**